src/server/qgsconfigcache.h

```diff
@@ -30,6 +31,38 @@
 #include "qgsproject.h"
 #include "qgsserversettings.h"
 
+#ifndef SIP_RUN
+
+class QgsConfigCache;
+
+/**
+ * \ingroup server
+ * \brief Abstract cache strategy for server configuration
+ * \since QGIS 3.24
+ */
+class SERVER_EXPORT QgsAbstractCacheStrategy
+{
+  public:
+    virtual QString name() const = 0;
+
+    /**
+     * Called when an entry is removed from cache.
+     * \param path The path of the project
+     */
+    virtual void entryRemoved( const QString &path ) = 0;
+
+    //! Called when an entry is inserted
+    virtual void entryInserted( const QString &path ) = 0;
+
+    //! Attache cache to this strategy
+    virtual void attach( QgsConfigCache *cache ) = 0;
```

Comment:
Can we use an reference to `QgsConfigCache` in this case?

Reply:
Done